### PR TITLE
fix: common schedule toggling incorrectly

### PR DESCRIPTION
**Pull request by emrysal**

### CODE REVIEW

File: packages/features/eventtypes/components/tabs/availability/EventAvailabilityTab.tsx [hunk 1]
```diff
@@ -439,42 +438,41 @@ const TeamAvailability = ({
...
+  const { useHostSchedulesForTeamEvent, toggleScheduleState } = useCommonScheduleState(eventType.schedule);
   return (
     <>
       <SettingsToggle
         checked={!useHostSchedulesForTeamEvent}
-        onCheckedChange={(checked) => {
-          setUseHostSchedulesForTeamEvent(!checked);
-          if (checked) {
-            if (Boolean(eventType.schedule)) resetField("schedule");
-            getValues("hosts").map((_, index) => {
```
Comment (emrysal) [inline packages/features/eventtypes/components/tabs/availability/EventAvailabilityTab.tsx:460]: This seems like a bad idea (on check, remove all host schedule Ids)

File: packages/features/eventtypes/components/tabs/availability/EventAvailabilityTab.tsx [hunk 2]
```diff
@@ -439,42 +438,41 @@ const TeamAvailability = ({
...
+
 const UseCommonScheduleSettingsToggle = ({ eventType, ...rest }: EventTypeScheduleProps) => {
   const { t } = useLocale();
-  const { setValue, resetField, getValues, watch } = useFormContext<FormValues>();
-
-  const [useHostSchedulesForTeamEvent, setUseHostSchedulesForTeamEvent] = useState(
-    !Boolean(eventType.schedule)
-  );
-
-  const watchHosts = watch("hosts");
```
Comment (emrysal) [inline packages/features/eventtypes/components/tabs/availability/EventAvailabilityTab.tsx:450]: Moved to child component

File: packages/features/eventtypes/components/tabs/availability/EventAvailabilityTab.tsx [hunk 3]
```diff
@@ -439,42 +438,41 @@ const TeamAvailability = ({
   );
 };
 
+const useCommonScheduleState = (initialScheduleId: number | null) => {
+  const { setValue } = useFormContext<FormValues>();
+  const [useHostSchedulesForTeamEvent, setUseHostSchedulesForTeamEvent] = useState(!initialScheduleId);
+  // Reset the main schedule
+  const clearMainSchedule = () => {
+    setValue("schedule", null, { shouldDirty: Boolean(initialScheduleId) });
```
Comment (emrysal) [inline packages/features/eventtypes/components/tabs/availability/EventAvailabilityTab.tsx:446]: Dirty if the initialScheduleId was set.

File: packages/features/eventtypes/components/tabs/availability/EventAvailabilityTab.tsx (hunk 1 above)
Comment (anikdhabal) [reply packages/features/eventtypes/components/tabs/availability/EventAvailabilityTab.tsx:460]: Yeah, this makes sense

**Review by anikdhabal — APPROVED** [review]
Great Changes
LGTM!!

**Pull request merged**